UI/window-projector.cpp

```diff
@@ -848,8 +848,51 @@ void OBSProjector::mousePressEvent(QMouseEvent *event)
 					SLOT(OpenWindowedProjector()));
 
 		} else if (!this->isMaximized()) {
+
+			auto resizeToCustomScaleAction = [this](double scale) {
+				ResizeToScale(scale);
+			};
+
+			auto resizeToScaleAction = [this](QAction *action) {
+				double scale = action->property("scale_amount")
+						       .toDouble();
+				ResizeToScale(scale);
+			};
+
 			popup.addAction(QTStr("ResizeProjectorWindowToContent"),
 					this, SLOT(ResizeToContent()));
+			QMenu *scaleMenu = new QMenu(
+				QTStr("ResizeProjectorWindowToSourceScale"));
+
+			double scalePresets[] = {0.5, 1, 1.5, 2, 2.5, 3, 4};
+
+			for (size_t i = 0;
+			     i < sizeof(scalePresets) / sizeof(double); i++) {
+				QAction *scale = new QAction(
+					QString("%1x").arg(scalePresets[i]),
+					this);
+				scale->setProperty("scale_amount",
+						   scalePresets[i]);
+				connect(scale, &QAction::triggered,
+					std::bind(resizeToScaleAction, scale));
+				scaleMenu->addAction(scale);
+			}
+			QDoubleSpinBox *customScale = new QDoubleSpinBox();
+			customScale->setMinimum(0.1);
+			customScale->setSuffix("x");
+			customScale->setSingleStep(0.5);
+			customScale->setValue(8.0);
+			customScale->setButtonSymbols(
+				QAbstractSpinBox::NoButtons);
+			connect(customScale,
+				(void (QDoubleSpinBox::*)(double)) &
+					QDoubleSpinBox::valueChanged,
+				resizeToCustomScaleAction);
```

Comment:
```suggestion
			connect(customScale, SIGNAL(valueChanged(double)), this,
				SLOT(ResizeToScale(double)));
```
This is easier to read, and more consistent with how we use this elsewhere.